pkg/finder/finder_test.go

```diff
@@ -0,0 +1,93 @@
+package finder
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/getsops/sops/v3"
+	"github.com/getsops/sops/v3/age"
+	"github.com/getsops/sops/v3/keys"
+	"github.com/getsops/sops/v3/kms"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidSopsFiles(t *testing.T) {
+	testDir := "testdata"
+
+	expectedResults := map[string]sops.Metadata{
+		"testdata/valid_sops_files/encrypted.env":  {},
+		"testdata/valid_sops_files/encrypted.ini":  {},
+		"testdata/valid_sops_files/encrypted.yaml": {},
+		"testdata/valid_sops_files/encrypted.yml":  {},
+		"testdata/valid_sops_files/encrypted.json": {},
+	}
+
+	// Loop through files in the testdata directory.
+	files, err := SopsFiles(testDir)
+	assert.NoError(t, err)
+	assert.Equal(t, len(expectedResults), len(files))
```

Comment:
You're only testing that the number of files matches, but not if the file paths and metadata are what you expect.